lib/galaxy/files/__init__.py

```diff
@@ -466,3 +473,7 @@ def app_vault(self):
     @property
     def file_sources(self):
         return self._kwd.get("file_sources")
+
+    @property
+    def anonymous(self) -> bool:
+        return self._kwd.get("username") is not None
```

Comment:
[posting this here; duplicate of what i posted to your fork]
Is it possible that username here is the empty string? Such values do exist in the database, 2 of them from today (on main).